generators/chipyard/src/main/scala/CoreManager.scala

```diff
@@ -0,0 +1,97 @@
+package chipyard
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+
+import chisel3._
+
+import freechips.rocketchip.config.{Parameters, Config, Field, View}
+import freechips.rocketchip.subsystem.{SystemBusKey, RocketTilesKey, RocketCrossingParams, RocketCrossingKey}
+import freechips.rocketchip.diplomacy.{LazyModule, ClockCrossingType, ValName}
+import freechips.rocketchip.diplomaticobjectmodel.logicaltree.LogicalTreeNode
+import freechips.rocketchip.rocket._
+import freechips.rocketchip.tile._
+
+import boom.common.{BoomTile, BoomTilesKey, BoomCrossingKey, BoomTileParams}
+import ariane.{ArianeTile, ArianeTilesKey, ArianeCrossingKey, ArianeTileParams}
+import chipsalliance.rocketchip.config.Parameters
+
+// Base trait for all third-party core entries
+sealed trait CoreEntryBase {
+  def tileParamsLookup(implicit p: Parameters): Seq[TileParams]
+  def updateWithFilter(view: View, p: Any => Boolean): PartialFunction[Any, Map[String, Any] => Any]
+  def instantiateTile(crossingLookup: (Seq[RocketCrossingParams], Int) => Seq[RocketCrossingParams], logicalTreeNode: LogicalTreeNode)
+    (implicit p: Parameters, valName: ValName): Seq[(TileParams, RocketCrossingParams, BaseTile)]
+}
+
+// Implementation of third-party core entries
+class CoreEntry[TileParamsT <: TileParams with Product: TypeTag, TileT <: BaseTile : TypeTag](
+  tk: Field[Seq[TileParamsT]],
```

Comment:
`tk` and `ck` seem like they can get better names

Reply:
NVM I see what they represent

Reply:
Im still of the opinion that it should be spelled out.